### PR TITLE
Fix S3908: Rule should not report issues on classes that implement in…

### LINKED ISSUE #550: Fix S3908: Rule should not report issues on classes that implement interfaces
Hello,

I have some classes that implement a specific interface ( System.Windows.Input.ICommand ). That ICommand defines an event of following signature:

            event EventHandler CanExecuteChanged;

As my class implements that interface, I have to implement that event as well.
However, Sonar reports that as an issue.

### Repro steps

Create a C# project and a class that implements the  System.Windows.Input.ICommand interface.

### Expected behavior

Rule S3908 is not violated.

### Actual behavior

Error window contains an entry that rule 3908 is violated.

### Known workarounds

Disable/suppress the rule temporarily

### Related information
* SonarLint for Visual Studio 2017 Version 3.2.1.1639
* Visual Studio Enterprise 2017 Version 15.2 (26430.15) Release

### PR DESCRIPTION
**Pull request by Evangelink**
…terfaces

Fix #550

### CODE REVIEW

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs [hunk 1]
```diff
@@ -39,29 +41,45 @@ public sealed class UseGenericEventHandlerInstances : SonarDiagnosticAnalyzer
...
-                        as INamedTypeSymbol;
-            if (eventHandlerType == null)
-            {
-                return;
-            }
+            var eventSymbol = analysisContext.SemanticModel.GetDeclaredSymbol(eventNode) as IEventSymbol;
 
-            if (!eventHandlerType.ConstructedFrom.Is(KnownType.System_EventHandler_TEventArgs))
+            if (eventSymbol != null &&
+                eventSymbol.GetOverriddenMember() == null &&
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs]: Use IsOverride instead.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs [hunk 2]
```diff
@@ -39,29 +41,45 @@ public sealed class UseGenericEventHandlerInstances : SonarDiagnosticAnalyzer
...
+            KnownType.System_EventHandler,
+            KnownType.System_EventHandler_TEventArgs
+        };
+
         protected override void Initialize(SonarAnalysisContext context)
         {
             context.RegisterSyntaxNodeActionInNonGenerated(
-               c => AnalyzeEventType(c, ((EventFieldDeclarationSyntax)c.Node).Declaration.Type),
-               SyntaxKind.EventFieldDeclaration);
+               c =>
```
Comment (michalb-sonar) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs]: Put "c" in previous line.

Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs]: We do it this way everywhere else.

File: sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs (hunk 1 above)
Comment (Evangelink) [inline sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseGenericEventHandlerInstances.cs]: Good catch!!

**Pull request merged**